### PR TITLE
Add /dd-org-promote command (org owner/admin only)

### PR DESCRIPTION
**Pull request by jnahian**
## Summary

- Adds `/dd-org-promote @user [TeamName]` — without `TeamName`, promotes the target to **organization admin**; with `TeamName`, promotes them to **team admin** in that team.
- Permission gated to active org `OWNER` or `ADMIN` (for both scopes).
- Reuses the existing `resolveTargetSlackUserId` helper, so mentions, raw Slack user IDs, and bare `@username`/`username` all work — same as the suspend commands. Registered without `stripFormatting` so the `<@U…|name>` mention token survives.
- Guards: rejects self-promotion, already-admin targets, org owners (already at the highest role), and inactive memberships.
- New service methods: `userService.promoteOrganizationMember()` and `teamService.promoteTeamMember()`.
- Updated `slack-app-manifest.json`, `README.md`, and `CHANGELOG.md`.

## Test plan

- [ ] Run `npm run manifest:update` to push the new slash command to Slack.
- [ ] As an org admin, `/dd-org-promote @member` — confirm target's `OrganizationMember.role` becomes `ADMIN`.
- [ ] As an org admin, `/dd-org-promote @member SomeTeam` — confirm target's `TeamMember.role` becomes `ADMIN` in that team.
- [ ] As a non-admin org member, `/dd-org-promote @member` — confirm permission error.
- [ ] `/dd-org-promote @already_admin` — confirm "already an admin" error.
- [ ] `/dd-org-promote @yourself` — confirm self-promotion error.
- [ ] `/dd-org-promote @member NonExistentTeam` — confirm team-not-found error.
- [ ] `/dd-org-promote @non_member SomeTeam` — confirm "not an active member of this team" error.

https://claude.ai/code/session_01LWagKtdbYZSZebpXGvHHJS

---
_Generated by [Claude Code](https://claude.ai/code/session_01LWagKtdbYZSZebpXGvHHJS)_

## Summary by CodeRabbit

* **New Features**
  * Added `/dd-org-promote `@user` [TeamName]` slash command to promote users to org or team admin. Accepts mentions, raw IDs, or bare usernames; preserves mention parsing. Prevents self-promotion and promoting existing owners/admins; shows helpful errors for invalid targets or teams.

* **Documentation**
  * Updated CHANGELOG, README and app manifest with command usage, scopes, and examples.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jnahian/daily-dose/pull/11?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/commands/team.js [hunk 1]
```diff
@@ -690,6 +690,89 @@ async function handleOrgSuspension({ command, ack, respond, client, suspend }) {
...
+  const updateResponse = await ackWithProcessing(
+    ack,
+    respond,
+    "Promoting member...",
+    command
+  );
+
+  try {
+    const parts = command.text.trim().split(/\s+/).filter(Boolean);
+
```
[the remark this answers is not in the document]
Comment (jnahian) [reply src/commands/team.js:703]: Skipping: `commandHelper.parseCommandText()` doesn't exist in `src/utils/commandHelper.js` (exports are `ackWithProcessing`, `removeFormatting`, `withFormattingRemoval`, `getChannelName`). CLAUDE.md references it as a convention but it's never been implemented — every existing slash-command handler (including the suspend commands this code mirrors: `handleTeamSuspension`, `handleOrgSuspension`) uses the same `command.text.trim().split(/\s+/).filter(Boolean)` pattern. Introducing the helper is a codebase-wide refactor and out of scope for this PR.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/commands/team.js [hunk 2]
```diff
@@ -690,6 +690,89 @@ async function handleOrgSuspension({ command, ack, respond, client, suspend }) {
...
+        text: `✅ <@${targetSlackUserId}> has been promoted to team admin in "${result.team.name}".`,
+      });
+      return;
+    }
+
+    const result = await userService.promoteOrganizationMember(
+      command.user_id,
+      targetSlackUserId,
+      client
+    );
```
[the remark this answers is not in the document]
Comment (jnahian) [reply src/commands/team.js:775]: Skipping: this would introduce a button-based interactive confirmation pattern that doesn't exist anywhere else in this codebase. The directly comparable destructive commands — `/dd-team-suspend`, `/dd-team-unsuspend`, `/dd-org-suspend`, `/dd-org-unsuspend` — all mutate state immediately without confirmation. Adding it only for `/dd-org-promote` would be inconsistent. The promotion is also reversible via a future demote/unsuspend equivalent and is gated to org owner/admin actors. Worth a follow-up PR if we want the confirmation pattern across all role-mutating commands, but out of scope here.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jnahian** [comment]
Pushed 93e114f addressing the cross-tenant team lookup flagged in review.

**What changed**
- `teamService.findTeamByName(teamName, organizationId = null)` — optional org filter; default-null preserves behavior for the 5 other callers, so this is a non-breaking signature extension.
- `/dd-org-promote` team-scope path now passes `adminOrg.id`, so the team must live in the actor's org. Eliminates the case where a same-named team in another org would resolve and route the permission check against the wrong org.
- Early reject when a team name is supplied but the actor has no org (previously would have silently fallen back to a global lookup via the null filter).
- "Team not found" → "Team not found in your organization" so the failure mode is honest.

**Not addressed in this commit** (from the review, deferred):
- `permissionHelper` consolidation — four call sites still do inline org-membership checks; would benefit from an `isOrgOwnerOrAdmin()` helper.
- `web/src/data/changelog.json` user-facing entry for `/dd-org-promote`.
- `getUserOrganization()` returning an arbitrary org for multi-org users (pre-existing across the suspend commands too).

Test plan unchanged from the PR description; the new edge case worth adding manually if you have a second workspace handy: an admin in Org A trying `/dd-org-promote @member SameNameTeam` where `SameNameTeam` only exists in Org B should now return "not found in your organization" instead of a permission error from the foreign org.
